tensorflow_probability/python/mcmc/transformed_kernel.py

```diff
@@ -109,6 +105,35 @@ def transformed_log_prob_fn(*state_parts):
     return tlp + sum(ldj_fn(state_parts, event_ndims))
   return transformed_log_prob_fn
 
+def _make_kernel_stack(kernel):
+  kernel_stack = [kernel]
+  while 'target_log_prob_fn' not in kernel.parameters:
+    if 'inner_kernel' not in kernel.parameters:
+      raise ValueError('"None of the nested `inner_kernel`s contains a '
+                       '`target_log_prob_fn`."')
+    kernel = kernel.inner_kernel
+    kernel_stack.append(kernel)
+  return kernel_stack
+
+def get_target_log_prob(kernel):
+  kernel_stack = _make_kernel_stack(kernel)
+  last_kernel = kernel_stack[len(kernel_stack) - 1]
+  return last_kernel.parameters['target_log_prob_fn']
+
+def update_target_log_prob(kernel, new_target_log_prob):
+  kernel_stack = _make_kernel_stack(kernel)
+  # get the last kernel and update to the new_target_log_prob
+  last_kernel = kernel_stack.pop().copy(target_log_prob_fn=new_target_log_prob)
+  last_kernel_kwargs = last_kernel.parameters
```

Comment:
Style: Do you need this here?  Can't you just access this field once in the place where you use `last_kernel_kwargs`, instead of maintaining another loop invariant?

Reply:
Good point; fixed!